scripts/add-token-cookie-param.js

```diff
@@ -0,0 +1,72 @@
+// The sendingRequest and responseReceived functions will be called for all requests/responses sent/received by ZAP, 
+// including automated tools (e.g. active scanner, fuzzer, ...)
+
+// Note that new HttpSender scripts will initially be disabled
+// Right click the script in the Scripts tree and select "enable"  
+
+// 'initiator' is the component the initiated the request:
+// 		1	PROXY_INITIATOR
+// 		2	ACTIVE_SCANNER_INITIATOR
+// 		3	SPIDER_INITIATOR
+// 		4	FUZZER_INITIATOR
+// 		5	AUTHENTICATION_INITIATOR
+// 		6	MANUAL_REQUEST_INITIATOR
+// 		7	CHECK_FOR_UPDATES_INITIATOR
+// 		8	BEAN_SHELL_INITIATOR
+// 		9	ACCESS_CONTROL_SCANNER_INITIATOR
+// 		10	AJAX_SPIDER_INITIATOR
+// For the latest list of values see the HttpSender class:
+// https://github.com/zaproxy/zaproxy/blob/main/zap/src/main/java/org/parosproxy/paros/network/HttpSender.java
+// 'helper' just has one method at the moment: helper.getHttpSender() which returns the HttpSender 
+// instance used to send the request.
+//
+// New requests can be made like this:
+// msg2 = msg.cloneAll() // msg2 can then be safely changed as required without affecting msg
+// helper.getHttpSender().sendAndReceive(msg2, false);
+// print('msg2 response=' + msg2.getResponseHeader().getStatusCode())
+
+function sendingRequest(msg, initiator, helper) {
+	
+    //  print('sendingRequest called for url=' + msg.getRequestHeader().getURI().toString())
+    var cookies = msg.getCookieParams();
+    //  print ("Cookies before update: " + cookies);
+
+    var updatedCookies = modifyCookies(cookies);
+     
+    msg.setCookieParams(updatedCookies);
```

Comment:
During my first try (with the faulty `hasAccessToken == 0` below), the JS script crashed with a error about a list being empty (I forgot to copy the precise error). I think this is caused by this line. 
My understanding is that `msg.setCookieParams()` expects a non-empty value.

As a neat-pick, to ensure that there is no crash even if there are no cookie (i.e.: the cookie list is empty *and* for some reason the user did not configure a cookieName/cookieVal) : 

```javascript
if ( updatedCookies ) {
  msg.setCookieParams(updatedCookies);
}
```